src/redux/web3listener.ts

```diff
@@ -1,11 +1,15 @@
 import { debounce, isEmpty } from 'lodash';
-import { web3Provider } from '../handlers/web3';
-import store from '../redux/store';
-import { multicallUpdateOutdatedListeners } from './multicall';
+import { ThunkDispatch } from 'redux-thunk';
+import { web3Provider } from '@rainbow-me/handlers/web3';
+import { multicallUpdateOutdatedListeners } from '@rainbow-me/redux/multicall';
+import store, { AppGetState, AppState } from '@rainbow-me/redux/store';
 import logger from 'logger';
 
 // -- Actions ---------------------------------------- //
-const updateMulticall = blockNumber => async (dispatch, getState) => {
+const updateMulticall = (blockNumber: number) => async (
+  dispatch: ThunkDispatch<AppState, unknown, never>,
```

Comment:
why ThunkDispatch instead of AppDispatch ?

Reply:
same for the other redux files

Reply:
This relates to https://github.com/rainbow-me/rainbow/pull/2498#discussion_r744327085, since there are other JS reducers that get included, the standard `AppDispatch` actually doesn't end up doing any type checking. Basically it resolves to `Dispatch<any>`. In this case I think `ThunkDispatch` instead of `Dispatch` because `multicallUpdateOutdatedListeners(blockNumber)` is being passed in instead of an object. For most of the other ones though, I used a `Dispatch` and specified which the valid actions are in order to get the type checking.